### DISCUSSION TITLE
Best approach for using existing database for evaluation context?

### DISCUSSION THREAD

**Discussion by amweiss** [flipt-io/flipt#1613] [Q&A]
Rather than having every request to flipt have to include all possible data in the request context, I was considering how we could have it hooked into an existing user database and only require the unique id for lookup of additional data.

It seems like hooking into the server is not possible today, and it would have to be deeper into somewhere like https://github.com/flipt-io/flipt/blob/de4da949a7e6aeddefb0c7153f1cf804efd96d49/internal/server/evaluator.go#L189 to make it have what is needed. So, my thought was building another service that wraps flipt that everything calls that only takes in the user id and then fetches whatever is needed for the other context items.

Am I missing anything? Server side it could be easier to pull in from other places in the context, but from client side I think we need to build something between the client and flipt.

**Comment by markphelps** [comment]
> Rather than having every request to flipt have to include all possible data in the request context, I was considering how we could have it hooked into an existing user database and only require the unique id for lookup of additional data.

You are literally reading my mind. Can you send me an email to `mark at flipt.io` ? I would love to share this RFC I've been working on about this exact thing and get your feedback. I think it is currently from the perspective of Flipt also running the user to context database, but I love the idea of making it pluggable for an org to run its own.
